src/main/java/org/taktik/icure/services/external/rest/v1/facade/MessageFacade.java

```diff
@@ -428,6 +429,18 @@ public Response setMessagesStatusBits(@PathParam("status") int status, ListOfIds
 		return ResponseUtils.ok(messageLogic.setStatus(messageIds.getIds(),status).stream().map(m->mapper.map(m,MessageDto.class)).collect(Collectors.toList()));
 	}
 
+	@ApiOperation(
+			value = "Set read status for given list of messages",
+			httpMethod = "PUT",
+			responseContainer = "List",
+			response = MessageDto.class
+	)
+	@PUT
+	@Path("/readstatus")
+	public Response setMessagesReadStatus(MessagesReadStatusUpdate data) throws MissingRequirementsException {
```

Comment:
Passing the request parameters in the body instead of in query parameters is a bit unusual compared to the rest of the API, but why not. Especially if it allows to accomodate for long lists of message IDs :+1:

Reply:
CC @antoinepairet